### PR TITLE
Noah mp snow

### PR DESCRIPTION
**Pull request by cenlinhe**
TYPE: enhancement, new feature

KEYWORDS: Noah-MP, snow, MPTABLE, parameter

SOURCE: Cenlin He (NCAR/RAL)

DESCRIPTION OF CHANGES:
Several Noah-MP snow-related aspects have been updated to improve snow simulations:
1. Add snow retention process at the snowpack bottom to improve streamflow modeling (also to be consistent 
with National Water Model updates)
2. Modify wind-canopy absorption coefficient (CWPVT) parameter values in MPTABLE to be vegetation dependent based on Goudriaan1977; previous values are incorrect.
3. Bring hard-coded snow emissivity of 1.0 to a global tunable MPTABLE parameter, and set to 0.95 to improve 
WRF surface temperature simulation
4. Bring hard-coded constant parameter (i.e., 2.5*z0, z0=0.002m) in snow cover formulation to a vegetation-
dependent tunable SCFFAC parameter in MPTABLE; values are currently optimized to improve SWE and surface 
albedo modeling by comparing with SNOTEL and MODIS observations.
5. Update MFSNO in snow cover formulation with optimized vegetation-dependent MFSNO values based on 
SNOTEL SWE & MODIS SCF & surface albedo comparison
6. Add wet-bulb temperature snow-rain partitioning scheme (OPT_SNF=5) based on Wang et al. 2019 GRL
7. Fix an issue for bulk leaf boundary layer resistance (RB) value range (>100) in RAGRB; Now using more realistic 
range (5~50)

LIST OF MODIFIED FILES: list of changed files
M       phys/module_sf_noahmpdrv.F
M       phys/module_sf_noahmplsm.F
M       run/MPTABLE.TBL

TESTS CONDUCTED: 
1. Conducted compilation and official em_real simulation tests with successful results in Cheyenne.
2. Passed Jenkins tests.

RELEASE NOTE: These Noah-MP updates include several snow-relevant enhancements and new features to improve snow simulations based on NCAR Noah-MP LSM team tests. The publication is still under development (He et al., 2021, in preparation). Snow improvements by this code update include: (1) new snow retention process improves streamflow modeling (tested with Noah-MP in National Water Model); (2) optimized parameter values for snow cover parameters (SCFFAC & MFSNO) improve snow water equivalent (SWE), snow depth, and surface albedo simulation; (3) new snow emissivity value improves WRF surface temperature simulation; (4) new wet-bulb temperature snow-rain partitioning scheme (OPT_SNF=5) improves SWE simulation see Wang et al. 2019 GRL; (5) enhanced flexibility of tuning model parameters by bringing the above-mentioned parameters from hard-coded values to MPTABLE; (6) Fix the incorrect values for wind-canopy absorption coefficient (CWPVT) parameter and the unrealistic value range of bulk leaf boundary layer resistance (RB).

### CODE REVIEW

**Comment by cenlinhe** [comment]
This is a PR separated from the original PR #1345.

**Review by dudhia — APPROVED** [review]

File: phys/module_sf_noahmplsm.F [hunk 1]
```diff
@@ -142,7 +142,8 @@ MODULE MODULE_SF_NOAHMPLSM
 		      !   2 -> BATS: when SFCTMP<TFRZ+2.2 
 		      !   3 -> SFCTMP < TFRZ
 		      !   4 -> Use WRF microphysics output
-
+                      !   5 -> Use wetbulb temperature (Wang et al., 2019 GRL) C.He, 12/18/2020
```
Comment (davegill) [inline phys/module_sf_noahmplsm.F:145]: @cenlinhe 
Line this up

File: phys/module_sf_noahmplsm.F [hunk 2]
```diff
@@ -1831,7 +1861,8 @@ SUBROUTINE ENERGY (parameters,ICE    ,VEGTYP ,IST    ,NSNOW  ,NSOIL  , & !in
      IF(SNOWH.GT.0.)  THEN
          BDSNO    = SNEQV / SNOWH
          FMELT    = (BDSNO/100.)**parameters%MFSNO
-         FSNO     = TANH( SNOWH /(2.5* Z0 * FMELT))
+         !FSNO     = TANH( SNOWH /(2.5* Z0 * FMELT))
```
Comment (davegill) [inline phys/module_sf_noahmplsm.F:1864]: Not required, as this is _your_ scheme. We find that removing old code is better than commenting it out. If this is left as an option for a developer, that is fine.

File: phys/module_sf_noahmplsm.F [hunk 3]
```diff
@@ -4333,7 +4364,8 @@ SUBROUTINE RAGRB(parameters,ITER   ,VAI    ,RHOAIR ,HG     ,TAH    , & !in
 
        TMPRB  = CWPC*50. / (1. - EXP(-CWPC/2.))
        RB     = TMPRB * SQRT(parameters%DLEAF/UC)
-       RB     = MAX(RB,100.0)
+       RB     = MIN(MAX(RB, 5.0),50.0) ! limit RB to 5~50
+!       RB     = MAX(RB,100.0)  ! typically RB<50,so this leads to constant RB of 100
 !       RB = 200
```
Comment (davegill) [inline phys/module_sf_noahmplsm.F:4369]: same idea, would it be better to remove old code that is not physically reasonable?

File: run/MPTABLE.TBL [hunk 4]
```diff
@@ -53,7 +53,11 @@
  HVB   =  1.00,  0.10,  0.10,  0.10,  0.10,  0.15,  0.05,  0.10,  0.10,  0.10,  11.5,  7.00,  8.00,  8.50,  10.0,  0.00,  0.05,  0.10,  0.00,  0.10,  0.10,  0.10,  0.10,  0.00,  0.10,  0.00,  0.00,
  DEN   =  0.01,  25.0,  25.0,  25.0,  25.0,  25.0,  100.,  10.0,  10.0,  0.02,  0.10,  0.28,  0.02,  0.28,  0.10,  0.01,  10.0,  0.10,  0.01,  1.00,  1.00,  1.00,  1.00,  0.00,  0.01,  0.01,  0.01,
  RC    =  1.00,  0.08,  0.08,  0.08,  0.08,  0.08,  0.03,  0.12,  0.12,  3.00,  1.40,  1.20,  3.60,  1.20,  1.40,  0.01,  0.10,  1.40,  0.01,  0.30,  0.30,  0.30,  0.30,  0.00,  0.01,  0.01,  0.01,
- MFSNO =  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,
+!MFSNO =  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,  2.50,
```
Comment (davegill) [inline run/MPTABLE.TBL:56]: @cenlinhe 
Does the module_sf_noahmplsm.F routine that reads this file handle really these comments?

Comment (davegill) [reply run/MPTABLE.TBL:56]: OK, these are namelists. Got it!

**Comment by cenlinhe** [comment]
@davegill @dudhia  Hi folks, have you merged my updates in this PR to the to-be-released version already? We found a mistake in the parameter value in the MPTABLE table. It is just the table value (run/MPTABLE.TBL). There is no change to the code files. We will try to update the table value using the correct values in one or two days (before Feb 21, 2021) and update the MPTABLE file in this PR. Please let me know if this works for you.

**Comment by dudhia** [comment]
It was approved but you should still be able to change it because it is not
merged.

File: phys/module_sf_noahmplsm.F (hunk 1 above)
Comment (cenlinhe) [reply phys/module_sf_noahmplsm.F:145]: will do.

File: phys/module_sf_noahmplsm.F (hunk 2 above)
Comment (cenlinhe) [reply phys/module_sf_noahmplsm.F:1864]: The reason to keep the original code is that we want users/developers to know which parameter has been changed here, and as you mentioned developers can always swtich between these two.

File: phys/module_sf_noahmplsm.F (hunk 3 above)
Comment (cenlinhe) [reply phys/module_sf_noahmplsm.F:4369]: will do.

**Pull request merged**